### PR TITLE
Remove Initialize() on Destroyer and PruneOptions

### PR DESCRIPTION
**Pull request by ash2k**

### CODE REVIEW

**Comment by seans3** [comment]
This looks good. I've just submitted a PR which necessitates a rebase. Once rebased, I'll merge it. Thanks for the cleanup.

File: pkg/apply/applier.go [hunk 1]
```diff
@@ -145,6 +143,9 @@ func (a *Applier) Run(ctx context.Context, invInfo inventory.InventoryInfo, obje
 			InfoHelper:   a.infoHelper,
 			Mapper:       mapper,
 			InvClient:    a.invClient,
+			Destroy:      false,
+			// TODO(seans): Remove this once Filter interface implemented.
```
Comment (seans3) [inline pkg/apply/applier.go:147]: The `LocalNamespaces` field has been removed in favor of the LocalNamespacesFilter.

File: pkg/apply/prune/prune.go [hunk 2]
```diff
@@ -73,6 +66,13 @@ type Options struct {
 	DryRunStrategy common.DryRunStrategy
 
 	PropagationPolicy metav1.DeletionPropagation
+
+	// True if we are destroying, which deletes the inventory object
+	// as well (possibly) the inventory namespace.
+	Destroy bool
+
+	// TODO(seans): Replace this with Filter interface to generalize prune skipping.
```
Comment (seans3) [inline pkg/apply/prune/prune.go:74]: The `LocalNamespaces` field should be removed.

File: pkg/apply/solver/solver.go [hunk 3]
```diff
@@ -45,6 +46,11 @@ type TaskQueueBuilder struct {
 	Factory      util.Factory
 	Mapper       meta.RESTMapper
 	InvClient    inventory.InventoryClient
+	// True if we are destroying, which deletes the inventory object
+	// as well (possibly) the inventory namespace.
+	Destroy bool
+	// TODO(seans): Replace this with Filter interface to generalize prune skipping.
+	LocalNamespaces sets.String
```
Comment (seans3) [inline pkg/apply/solver/solver.go:53]: This `LocalNamespaces` field should be removed.

File: pkg/apply/task/prune_task.go [hunk 4]
```diff
@@ -25,6 +26,11 @@ type PruneTask struct {
 	Filters           []filter.ValidationFilter
 	DryRunStrategy    common.DryRunStrategy
 	PropagationPolicy metav1.DeletionPropagation
+	// True if we are destroying, which deletes the inventory object
+	// as well (possibly) the inventory namespace.
+	Destroy bool
+	// TODO(seans): Replace this with Filter interface to generalize prune skipping.
+	LocalNamespaces sets.String
```
Comment (seans3) [inline pkg/apply/task/prune_task.go:33]: This `LocalNamespaces` field should be removed.

**Review by seans3 — COMMENTED** [review]
The rebasing kept a field which should be removed--`LocalNamespaces`. The field should be removed.

**Comment by ash2k** [comment]
Ops, I thought I removed it, sorry.

**Review by seans3 — APPROVED** [review]
/lgtm

**Pull request merged**